js/views/modals/listingDetail/Listing.js

```diff
@@ -308,6 +333,30 @@ export default class extends BaseModal {
     }
   }
 
+  randomizeMoreListings(cl) {
+    if (!cl.length) {
+      return [];
+    }
+
+    const listings = cl.models.filter(md =>
+      md.get('slug') !== this.model.get('slug'));
+    const tot = listings.length < 8 ? listings.length : 8;
+    const results = [];
+
+    for (let i = 0; i < tot; i++) {
+      let model;
+      do {
+        const index = randomInt(0, tot - 1);
```

Comment:
Do we want to use a random number ceiling of tot -1 or listings.length -1?

If it's tot -1 it will always be random within the first 8 or less results, won't it? If it was listings.length -1 it would be randomly selected from all the results, which might be nicer if the store has a lot of listings.